packages/gatsby-tinacms-mdx/src/useMdxForm.tsx

```diff
@@ -0,0 +1,226 @@
+  const valuesOnDisk = useMemo(
+    () => ({
+      fileRelativePath: mdx.fileRelativePath,
+      frontmatter: mdx.frontmatter,
+      rawMarkdownBody: mdx.rawMarkdownBody,
+      rawFrontmatter: JSON.parse(mdx.rawFrontmatter)
+    }),
+    [mdx.rawFrontmatter, mdx.rawMarkdownBody]
+  )
+
+  /**
+   * The state of the Mdx, generated from the contents of the
+   * Markdown file at the HEAD of this git branch.
+   */
+  /* eslint-disable-next-line react-hooks/rules-of-hooks */
+  const [valuesInGit, setValuesInGit] = React.useState()
+  /* eslint-disable-next-line react-hooks/rules-of-hooks */
+  React.useEffect(() => {
+    cms.api.git
+      .show(id) // Load the contents of this file at HEAD
+      .then((git: any) => {
+        // Parse the content into the Mdx data structure and store it in state.
+        const { content: rawMarkdownBody, data: rawFrontmatter } = matter(
+          git.content
+        )
+        setValuesInGit({ ...valuesOnDisk, rawFrontmatter, rawMarkdownBody })
+      })
+      .catch((e: any) => {
+        console.log('FAILED', e)
+      })
+  }, [id])
+
+  /**
+   * The list of Field definitions used to generate the form.
+   */
+  /* eslint-disable-next-line react-hooks/rules-of-hooks */
+  const fields = React.useMemo(() => {
+    let fields = formOverrrides.fields || generateFields(valuesOnDisk)
+    fields = fields.map(field => {
+      /**
+       * Treat the field.name prefix `frontmatter` as an alias to
+       * `rawFrontmatter`. This is to make defining fields more intuitive.
+       */
+      if (
+        field.name === 'frontmatter' ||
+        field.name.startsWith('frontmatter.')
+      ) {
+        return {
+          ...field,
+          name: field.name.replace('frontmatter', 'rawFrontmatter')
+        }
+      }
+      return field
+    })
+
+    return fields
+  }, [formOverrrides.fields])
+
+  /* eslint-disable-next-line react-hooks/rules-of-hooks */
+  const [, form] = useForm(
+    {
+      label,
+      id,
+      initialValues: valuesInGit,
+      fields,
+      onSubmit(data) {
+        return cms.api.git.onSubmit!({
+          files: [data.fileRelativePath],
+          message: data.__commit_message || 'Tina commit',
+          name: data.__commit_name,
+          email: data.__commit_email
+        })
+      },
+      reset() {
+        return cms.api.git.reset({ files: [id] })
+      },
+      actions
+    },
+    // The Form will be updated if these values change.
+    {
+      label,
+      fields,
+      values: valuesOnDisk
+    }
+  )
+
+  /* eslint-disable-next-line react-hooks/rules-of-hooks */
+  const writeToDisk = React.useCallback(formState => {
+    cms.api.git.onChange!({
+      fileRelativePath: formState.values.fileRelativePath,
+      content: toMarkdownString(formState.values)
+    })
+  }, [])
+
+  /* eslint-disable-next-line react-hooks/rules-of-hooks */
+  useWatchFormValues(form, writeToDisk)
+
+  return [mdx, form]
+}
+
+export function useLocalMdx(
```

Comment:
This is small, but I think naming this `useLocalMdxForm` will be more cohesive with the other packages. Along with `useGlobalMdxForm`

Reply:
Totally agree with you. Consistency is key. Great catch! Just fixed.